lib/ansible/modules/cloud/azure/azure_rm_acr.py

```diff
@@ -0,0 +1,342 @@
+                        registry_name=self.name,
+                        registry=Registry(
+                            location=self.location,
+                            sku=Sku(
+                                name=self.sku
+                            ),
+                            tags=self.tags,
+                            admin_user_enabled=self.admin_user_enabled
+                        )
+                    )
+                else:
+                    poller = self.containerregistry_mgmt_client.registries.update(
+                        resource_group_name=self.resource_group,
+                        registry_name=self.name,
+                        registry_update_parameters=RegistryUpdateParameters(
+                            sku=Sku(
+                                name=self.sku
+                            ),
+                            tags=self.tags,
+                            admin_user_enabled=self.admin_user_enabled
+                        )
+                    )
+                response = self.get_poller_result(poller)
+            else:
+                response = None
+        except CloudError as exc:
+            self.log('Error attempting to create the ACR instance.')
+            self.fail("Error creating the ACR instance: {0}".format(str(exc)))
+        return create_acr_dict(response)
+
+    def delete_acr(self):
+        '''
+        Deletes the specified container registry in the specified subscription and resource group.
+
+        :return: True
+        '''
+        self.log("Deleting the ACR instance {0}".format(self.name))
+        try:
+            self.containerregistry_mgmt_client.registries.delete(self.resource_group, self.name).wait()
+        except CloudError as e:
+            self.log('Error attempting to delete the ACR instance.')
+            self.fail("Error deleting the ACR instance: {0}".format(str(e)))
+
+        return True
+
+    def get_acr(self):
+        '''
+        Gets the properties of the specified container registry.
+
+        :return: deserialized ACR state dictionary
+        '''
+        self.log("Checking if the ACR instance {0} is present".format(self.name))
+        found = False
+        try:
+            response = self.containerregistry_mgmt_client.registries.get(self.resource_group, self.name)
+            found = True
+            self.log("Response : {0}".format(response))
+            self.log("ACR instance : {0} found".format(response.name))
+        except CloudError as e:
+            self.log('Did not find the ACR instance.')
+        if found is True:
+            return create_acr_dict(response)
+        else:
+            return False
+    
+    @property
+    def containerregistry_mgmt_client(self):
+        self.log('Getting container registry mgmt client')
+        if not self._containerregistry_mgmt_client:
+            self._containerregistry_mgmt_client = ContainerRegistryManagementClient(
```

Comment:
Two minor cents:
- After you pull the latest changes from `devel`, you can use the `get_mgmt_svc_client` method to instantiate your client. User agent will be automatically added.
- You may want to add resource management base url like [here](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/azure_rm_common.py#L734) to support multiple clouds (like China, Germany etc.).

Reply:
good stuff